contracts/DistributionCreator.sol

```diff
@@ -650,98 +443,179 @@ contract DistributionCreator is UUPSHelper, ReentrancyGuardUpgradeable {
-        extensiveParams.rewardTokenDecimals = IERC20Metadata(distribution.rewardToken).decimals();
-    }
-
-    /// @notice Tries to fetch extra data about the parameters in a distribution
-    function tryGetExtensiveDistributionParameters(
-        DistributionParameters memory distribution
-    ) external returns (bool success, ExtensiveDistributionParameters memory extensiveParams) {
-        (bool callSuccess, bytes memory returndata) = address(this).delegatecall(
-            abi.encodeWithSelector(DistributionCreator.getExtensiveDistributionParameters.selector, distribution)
-        );
-        success = callSuccess;
-        if (success) {
-            extensiveParams = abi.decode(returndata, (ExtensiveDistributionParameters));
+        assembly {
+            mstore(activeRewards, length)
         }
-        return (success, extensiveParams);
+        return (activeRewards, i);
     }
 
-    /// @notice Gets the list of all the distributions for `uniV3Pool` that have been active between `epochStart` and `epochEnd` (excluded)
-    /// @dev If the `uniV3Pool` parameter is equal to 0, then this function will return the distributions for all pools
-    function _getPoolDistributionsBetweenEpochs(
-        address uniV3Pool,
+    /// @notice Internal version of `getDistributionsBetweenEpochs`
+    function _getDistributionsBetweenEpochs(
         uint32 epochStart,
         uint32 epochEnd,
         uint32 skip,
         uint32 first
-    ) internal view returns (ExtensiveDistributionParameters[] memory, uint256) {
+    ) internal view returns (DistributionParameters[] memory, uint256) {
         uint256 length;
         uint256 distributionListLength = distributionList.length;
         uint256 returnSize = first > distributionListLength ? distributionListLength : first;
-        ExtensiveDistributionParameters[] memory activeRewards = new ExtensiveDistributionParameters[](returnSize);
+        DistributionParameters[] memory activeRewards = new DistributionParameters[](returnSize);
         uint32 i = skip;
         while (i < distributionListLength) {
-            DistributionParameters memory distribution = distributionList[i];
-            if (
-                _isDistributionLiveBetweenEpochs(distribution, epochStart, epochEnd) &&
-                (uniV3Pool == address(0) || distribution.uniV3Pool == uniV3Pool)
-            ) {
-                (bool success, ExtensiveDistributionParameters memory extensiveParams) = IDistributionCreator(
-                    address(this)
-                ).tryGetExtensiveDistributionParameters(distribution);
-                if (success) {
-                    activeRewards[length] = extensiveParams;
-                    length += 1;
-                }
+            DistributionParameters memory d = distributionList[i];
+            if (d.epochStart + d.numEpoch * HOUR > epochStart && d.epochStart < epochEnd) {
```

Comment:
why don't we need the `try` function anymore?

Reply:
because we do not do external calls here anymore